### ISSUE TITLE
Update Mouse Phenotypes detail view

### ISSUE THREAD

**Issue by andrewhercules** [opentargets/platform#1639]
After reviewing the work on the `target` ETL in #1313, the query for the Mouse Phenotypes detail view is expected to change significantly — details TBC.

**Comment by andrewhercules** [comment]
Ticket on hold pending further upstream data input and processing work in #1642

**Comment by tskir** [comment]
### Changes required to accommodate `mousePhenotypes` schema updates (https://github.com/opentargets/platform/issues/1471)

Backend related changes are described in https://github.com/opentargets/platform/issues/1641#issuecomment-893386335. The frontend related changes are:
* [ ] **Link to the mouse model information.**
  + The identifier will be provided in the new `biologicalModelId` column, e.g. _MGI:4421538._ This can be used to construct a link: http://www.informatics.jax.org/allele/genoview/MGI:4421538. This link should be applied to the  values in the “Allelic composition” column.
  + Note: MGI gene identifier (`targetInModelMgiId`, e.g. _MGI:87859_) shares the same `MGI:` namespace, but these are distinct concepts and should not be confused.
* [ ] **Amend gene representation to support two identifier systems.**
  + Currently, only one gene identifier is supplied through `mouse_gene_id`, e.g. _MGI:87859._ The “Mouse gene” column is a name pointing to an automatically generated link such as http://www.informatics.jax.org/marker/MGI:87859.
  + However, there is another widely used identifier system for mouse genes: Ensembl Mouse IDs. They will also be provided in the new schema through the `targetInModelEnsemblId` field, e.g. _ENSMUSG00000026842._ It can also be used to generate a link pointing to http://www.ensembl.org/id/ENSMUSG00000026842.
  + New UI representation needs to be considered to support having two links per row.
  + In rare cases, this field will be empty (not present). This is expected, because not all MGI gene identifiers have cross-links to Ensembl identifiers.
* [ ] **“Phenotype category” and “Phenotype label” values should be clickable links leading to MP ontology pages.**
  + For example, if `modelPhenotypeClassId` = _MP:0005385_ and `modelPhenotypeClassLabel` = _cardiovascular system phenotype,_ then the displayed label should be a link pointing to http://www.informatics.jax.org/vocab/mp_ontology/MP:0005385.

**Comment by andrewhercules** [comment]
Thank you @tskir - once the data is ready, I'll ask the front-end team to implement 👍

**Comment by andrewhercules** [comment]
With the new `target` ETL pipeline and API response from @tskir and @JarrodBaker, we now have new insights to display in the Mouse Phenotypes section.

```
query targetInfo {
  target(ensemblId: "ENSG00000145777") {
    id
    approvedSymbol
    mousePhenotypes{
      biologicalModelAllelicComposition
      biologicalModelGeneticBackground
      biologicalModelId
      literature
      modelPhenotypeClassId
      modelPhenotypeClassLabel
      modelPhenotypeId
      modelPhenotypeLabel
      targetInModel
      targetInModelEnsemblId
      targetInModelMgiId
    }
  }
}
```
Sample query: [TSLP](https://api-beta-dot-open-targets-eu-dev.ew.r.appspot.com/api/v4/graphql/browser?query=query%20targetInfo%20%7B%0A%20%20target%28ensemblId%3A%20%22ENSG00000145777%22%29%20%7B%0A%20%20%20%20id%0A%20%20%20%20approvedSymbol%0A%20%20%20%20mousePhenotypes%7B%0A%20%20%20%20%20%20biologicalModelAllelicComposition%0A%20%20%20%20%20%20biologicalModelGeneticBackground%0A%20%20%20%20%20%20biologicalModelId%0A%20%20%20%20%20%20literature%0A%20%20%20%20%20%20modelPhenotypeClassId%0A%20%20%20%20%20%20modelPhenotypeClassLabel%0A%20%20%20%20%20%20modelPhenotypeId%0A%20%20%20%20%20%20modelPhenotypeLabel%0A%20%20%20%20%20%20targetInModel%0A%20%20%20%20%20%20targetInModelEnsemblId%0A%20%20%20%20%20%20targetInModelMgiId%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D%0A)

Can we please update the summary widget and detail view?

### Summary widget

- [ ] Please show the number of unique phenotypes returned by the API

This number can be generated using the `mousePhenotypes` array, which returns a list of objects that have a `modelPhenotypeLabel` field. Please use this field and count the number of unique array entries. 

Note: we cannot use the length of the `mousePhenotypes` array as some `modelPhenotypeLabel` entries can appear more than once as they can belong to different categories. For example, with TSLP "colitis" appears in three different categories — "digestive/alimentary phenotype", "immune system phenotype", and "mammalian phenotype".

### Detail view

| Mouse gene | Category | Phenotype | Allelic composition | Publications |
| --- | --- | --- | --- | --- |
| `targetInModel` | `modelPhenotypeClassLabel` | `modelPhenotypeLabel` | `biologicalModelAllelicComposition` <br /> `biologicalModelGeneticBackground`| `literature`
| [Tslp](http://identifiers.org/MGI:1855696) | [hematopoietic system phenotype](http://identifiers.org/MP:0005397) | [abnormal immunoglobulin level](http://identifiers.org/MP:0002490) |[ Tslp<tm1(KOMP)Vlcg>/Tslp<tm1(KOMP)Vlcg> <br /> involves: C57BL/6 * C57BL/6NTac](http://identifiers.org/MGI:5438831) | [22585305](url) |

- [ ] In the "Mouse gene" column, please hyperlink the `targetInModel` value to MGI using the `targetInModelMgiId` and identifiers.org — `https://identifiers.org/` + `targetInModelMgiId`

- [ ] In the "Category" column, please hyperlink the `modelPhenotypeClassLabel` to OLS using the `modelPhenotypeClassId` and identifiers.org — `https://identifiers.org/` + `modelPhenotypeClassId`

- [ ] In the "Phenotype" column, please hyperlink the `modelPhenotypeLabel` to OLS using the `modelPhenotypeId` and identifiers.org — `https://identifiers.org/` + `modelPhenotypeId`

- [ ] In the "Allelic composition" column, please [format the data](https://github.com/opentargets/platform-app/blob/main/src/components/MouseModelAllelicComposition.js) as we currently do in production for both Mouse Phenotypes and PhenoDigm data. Please also hyperlink both the `biologicalModelAllelicComposition` and `biologicalModelGeneticBackground` values to MGI using `biologicalModelId` and identifiers.org — `https://identifiers.org/` + `biologicalModelId`

- [ ] In the "Publications" column, please use the publications drawer component to show the citation/abstract. Note that phenotypes can have one or more publications — if there is one publication, show the PubMed ID and if there are more than one publication, show "`n` publications" where `n` is the number of the publications in the `literature` array.
